app.go

```diff
@@ -308,11 +336,48 @@ func New(opts ...Option) *App {
 
 	if err := app.executeInvokes(); err != nil {
 		app.err = err
+		var b bytes.Buffer
+
+		if dig.CanVisualizeError(err) {
+			dig.Visualize(app.container, &b, dig.VisualizeError(err))
+		}
+
+		app.errorHooks.onError(wrappedErr{
+			graph: b.String(),
+			err:   err,
+		})
 	}
 
 	return app
 }
 
+type wrappedErr struct {
```

Comment:
how about visualizedError or errorWithGraph?

Reply:
`errorWithGraph` sounds good, I'll rename it to that 😄